### PR TITLE
ci: Add standalone macOS build support

### PR DESCRIPTION
**Pull request by jamilbk**
The CI swift workflow needs to be updated to accommodate the macOS standalone build. This required a decent amount of refactoring to make the Apple build process more maintainable.

Unfortunately this PR ended up being a giant ball of yarn where pulling on one thread tended to unravel things elsewhere, since building the Apple artifacts involve multiple interconnected systems. Combined with the slow iteration of running in CI, I wasn't able to split this PR into easier to digest commits, so I've annotated the PR as much as I can to explain what's changed.

The good news is that Apple release artifacts can now be easily built from a developer's machine with simply `scripts/build/macos-standalone.sh`. The only thing needed is the proper provisioning profiles and signing certs installed.

Since this PR is so big already, I'll save the swift/apple/README.md updates for another PR.

### CODE REVIEW

File: .github/workflows/_swift.yml
Comment (jamilbk) [inline .github/workflows/_swift.yml:9]: This was updated to match our development machines which reduces the build configuration drift between the two.

File: .github/workflows/_swift.yml
Comment (jamilbk) [inline .github/workflows/_swift.yml:14]: Updated to be a recent version. There were a few subtle differences in how provisioning profiles and certs are stored/loaded in Xcode 16 that will be noted where appropriate.

File: .github/workflows/_swift.yml
Comment (jamilbk) [inline .github/workflows/_swift.yml]: This was refactored to be a matrix for each distributable, with the upload process carved out into another script.

secrets are not available from the `strategy` context, otherwise I would have added them here and not in the build steps below.

File: .github/workflows/_swift.yml
Comment (jamilbk) [inline .github/workflows/_swift.yml:56]: Not every matrix job requires every secret, but doing it like this simplifies the build scripts.

The overall goal here was to not require any ENV vars when building locally, and override appropriate ones from CI if we're building on a runner.

File: scripts/build/dmg_background.png
Comment (jamilbk) [inline scripts/build/dmg_background.png:1]: Embedding this was somewhat of a pain, but I think it's important. Many Mac users simply try to (mistakenly) run their apps directly from the readonly disk image.

For many apps that works just fine, but for ours it won't due to the network extension. So I spent some time educating the user just in case, to avoid future support burden here.

File: scripts/build/ios-appstore.sh
Comment (jamilbk) [inline scripts/build/ios-appstore.sh:1]: Build locally with simply

`scripts/build/ios-appstore.sh`

Requires the provisioning profiles to be installed (download from developer portal then drag on Xcode) and the certs (incl signing key) to be installed (from 1Pass)

File: scripts/build/ios-appstore.sh
Comment (jamilbk) [inline scripts/build/ios-appstore.sh:11]: These are vitally important to get right. Each time anything related to certs, capabilities, profiles, in the developer portal changes, these will change.

File: scripts/build/ios-appstore.sh
Comment (jamilbk) [inline scripts/build/ios-appstore.sh:32]: Build configuration was moved from xcconfig files to here. I thought this is more explicit and consolidates all of our build context to one place for easier maintainability.

File: scripts/build/ios-appstore.sh
Comment (jamilbk) [inline scripts/build/ios-appstore.sh:48]: I decided to inline this file to consolidate build context in this script. This file is not used in debug builds.

File: scripts/build/ios-appstore.sh
Comment (jamilbk) [inline scripts/build/ios-appstore.sh:79]: ARCHIVE_PATH is shared between build and uploader so the upload script knows what to upload.

File: scripts/build/lib.sh
Comment (jamilbk) [inline scripts/build/lib.sh:27]: This is new; needed in newer versions of Xcode for it to find our codesigning certs.

File: scripts/build/lib.sh
Comment (jamilbk) [inline scripts/build/lib.sh:80]: This is now needed to prevent Xcode from prompting the user with a UI dialog to unlock the keychain to find the certs. This seems to have been a workaround [for some time](https://stackoverflow.com/questions/39868578/security-codesign-in-sierra-keychain-ignores-access-control-settings-and-ui-p), but only recently the GitHub actions runners require it.

File: scripts/build/macos-standalone.sh
Comment (jamilbk) [inline scripts/build/macos-standalone.sh:91]: Yep, AppleScript is the only way to do this in CI 🤠

File: scripts/build/macos-standalone.sh
Comment (jamilbk) [inline scripts/build/macos-standalone.sh]: This var is true when we merge to main or run this workflow from a manual trigger.

File: scripts/upload/app-store-connect.sh
Comment (jamilbk) [inline scripts/upload/app-store-connect.sh]: Slight change here to use a base64-encoded version of the API key. It is already in a PEM format, but the downstream build-rust.sh script was barfing on its contained newlines, so this fixes that.

File: scripts/upload/github.sh
Comment (jamilbk) [inline scripts/upload/github.sh:1]: We do this from quite a few other places in CI. I'll look into reusing this script for those.

File: swift/apple/.gitignore
Comment (jamilbk) [inline swift/apple/.gitignore:8]: Ok so this change simplifies things nicely. Instead of defining multiple xcconfigs and moving them into place just before build, we simply specify the build vars on the command directly. So we can now commit a single xcconfig and use that for development and the release build scripts will override what vars they need.

File: swift/apple/Firezone.xcodeproj/project.pbxproj
Comment (jamilbk) [inline swift/apple/Firezone.xcodeproj/project.pbxproj:1]: Simplifications from the removal of xcconfigs.

File: swift/apple/Firezone/ExportOptions.plist
Comment (jamilbk) [inline swift/apple/Firezone/ExportOptions.plist:1]: This file has been inlined in the ios build script.

File: scripts/build/dmg_background.png
Comment (thomaseizinger) [inline scripts/build/dmg_background.png:1]: I've seen it for several MacOS apps to have a somewhat playful thing here. I think JetBrains for example has something like:

```
(<logo.png>).install()
```

It would probably be quite easy to just once-off hire a designer to come up with a cool idea here. Maybe a tunnel that you can drag the Firezone logo through?

File: scripts/build/lib.sh
Comment (thomaseizinger) [inline scripts/build/lib.sh]:
```suggestion
    local profile="$1"
    local profiles_path="$2"
    local profile_file="$3"
```

I think this order would make more sense, perhaps also don't build the path within the function?

File: scripts/build/lib.sh
Comment (thomaseizinger) [reply scripts/build/lib.sh:80]: Perhaps add this information as a source-code comment as to why we are doing this.

File: scripts/build/lib.sh
Comment (thomaseizinger) [inline scripts/build/lib.sh]:
```suggestion
function set_current_project_version() {
```

File: scripts/build/macos-appstore.sh
Comment (thomaseizinger) [inline scripts/build/macos-appstore.sh:39]: Are these ENV variables? It is odd that they are defined as part of the build command and not at the very front like ENV vars for regular commands.

File: scripts/build/macos-standalone.sh
Comment (thomaseizinger) [inline scripts/build/macos-standalone.sh]: This could reuse the other function in `lib` if you could just pass a single output path.

File: scripts/build/macos-standalone.sh
Comment (thomaseizinger) [inline scripts/build/macos-standalone.sh]: A few _minutes_? What does this do? Wait for manual approval of an Apple support person?

File: scripts/upload/app-store-connect.sh
Comment (thomaseizinger) [reply scripts/upload/app-store-connect.sh]: Could also reuse that utility decode function used in `lib.sh` when building. Not sure if it is worth consolidating though.

File: scripts/upload/github.sh
Comment (thomaseizinger) [inline scripts/upload/github.sh:1]: Maybe call this script `github-release` because we also upload stuff as GitHub artifacts.

File: .github/workflows/_swift.yml
Comment (thomaseizinger) [inline .github/workflows/_swift.yml]: So all env variables here are by convention upper-cased? Don't feel strongly about it but the convention seems a bit unnecessary.

Also would prefer all "scripts" to be grouped together.

**Review by thomaseizinger — APPROVED** [review]
Great job! I think the structure makes a lot of sense. ~Can we extend `swift/apple/README.md` to document how to use the scripts?~

Edit: Just read your PR description.

File: scripts/build/lib.sh
Comment (jamilbk) [reply scripts/build/lib.sh]: 👍

File: scripts/build/lib.sh
Comment (jamilbk) [reply scripts/build/lib.sh:80]: 👍

File: scripts/build/lib.sh
Comment (jamilbk) [reply scripts/build/lib.sh]: Apple calls these `build version`, updated this to `set_project_build_version`.

File: scripts/build/macos-appstore.sh
Comment (jamilbk) [reply scripts/build/macos-appstore.sh:39]: Yes and no. Xcode will pass them as env vars to downstream scripts when appropriate, but they won't persist beyond that.

I thought this way reduced the chance these env vars can affect the `productbuild` command just below and others.

I'm probably a little traumatized from my experience with env vars and Xcode / cargo downstream effects 😆

File: scripts/build/macos-standalone.sh
Comment (jamilbk) [reply scripts/build/macos-standalone.sh]: Yeah I made this a `base64_decode` function for reuse. Probably a good idea since it deals with secrets and we don't want to fudge that up.

File: scripts/build/macos-standalone.sh
Comment (jamilbk) [reply scripts/build/macos-standalone.sh]: Yes, the API is quite unfriendly: https://developer.apple.com/documentation/security/customizing-the-notarization-workflow#Check-the-status-of-your-request

> the notarization process typically takes less than an hour.

I was thinking we can start with this and see how long the waits are. If they _really_ take an hour we might need the notarization step to happen in a separate workflow or job.

File: scripts/upload/app-store-connect.sh
Comment (jamilbk) [reply scripts/upload/app-store-connect.sh]: 👍

File: scripts/upload/github.sh
Comment (jamilbk) [reply scripts/upload/github.sh:1]: 👍

File: scripts/build/dmg_background.png
Comment (jamilbk) [reply scripts/build/dmg_background.png:1]: Hm yeah, though generating this is a little tricky and requires some trial and error since the window coordinates didn't seem to line up to Figma's pixel coordinates. I.e. getting the icons placed onto the PNG in the right place in the script might make the process a little cumbersome for an external designer.

I'll spend a few more minutes to see if I can make it a little snazzier.

File: .github/workflows/_swift.yml
Comment (jamilbk) [reply .github/workflows/_swift.yml]: 👍

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**
